core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java

```diff
@@ -134,6 +118,51 @@ protected RowTable() {
     }
   }
 
+  /** Helper class for the test for [CALCITE-6764] below. */
+  private static class RowTable extends EmptyTable {
+    protected RowTable() {
+      super();
+    }
+
+    @Override public RelDataType getRowType(RelDataTypeFactory typeFactory) {
+      final PairList<String, RelDataType> columnDesc = PairList.withCapacity(1);
+      // Schema contains a column whose type is MAP<VARCHAR, ROW(VARCHAR)>, but
+      // the ROW type can be nullable.  This can conceivably be created by a
+      // declaration such as
+      // CREATE TABLE T(p MAP<VARCHAR, ROW(k VARCHAR)>);
+      final RelDataType colType =
+          typeFactory.createMapType(typeFactory.createSqlType(SqlTypeName.VARCHAR),
+            new RelRecordType(
+                StructKind.PEEK_FIELDS,
+                  ImmutableList.of(
+                      new RelDataTypeFieldImpl("K", 0,
+                          typeFactory.createSqlType(SqlTypeName.VARCHAR))), true));
+      columnDesc.add("P", colType);
+      return typeFactory.createStructType(columnDesc);
+    }
+  }
+
+  /** Helper class for the test for [CALCITE-6764] below. */
+  private static class RowTable2 extends EmptyTable {
```

Comment:
Nit: the difference between `RowTable` and `RowTable2` seems to be in the nullability of the map composing the row type, can we make this explicit by calling them `NullableRowTable` and `RowTable` or something along this line?